be/ut/tx_bulk.c

```diff
@@ -357,24 +391,37 @@ static void be_ut_tx_bulk_state_test_run(struct be_ut_tx_bulk_state  *tbs,
 					 struct be_ut_tx_bulk_be_cfg *be_cfg,
                                          bool                         success)
 {
-	struct m0_be_tx_bulk_cfg    tb_cfg = {
-		.tbc_next   = &be_ut_tx_bulk_state_next,
-		.tbc_credit = &be_ut_tx_bulk_state_credit,
-		.tbc_do     = &be_ut_tx_bulk_state_do,
+	struct m0_be_tx_bulk_cfg tb_cfg = {
+		.tbc_q_cfg                 = {
+			.bqc_q_size_max       = BE_UT_TX_BULK_Q_SIZE_MAX,
+			.bqc_producers_nr_max = 1,
+			.bqc_consumers_nr_max = 0x100,  /* XXX */
```

Comment:
`be-ut:tx_bulk-usecase` UT is failing for `bqc_producers_nr_max = 1`  and `bqc_consumers_nr_max = 1` 
```
#0  0x00007fffeb0ea337 in raise () from /lib64/libc.so.6
#1  0x00007fffeb0eba28 in abort () from /lib64/libc.so.6
#2  0x00007fffec90be85 in m0_arch_panic (c=c@entry=0x7fffeccf8ba0 <__pctx.7470>, ap=ap@entry=0x7fffea0b1a48) at lib/user_space/uassert.c:131
#3  0x00007fffec8fd3c4 in m0_panic (ctx=ctx@entry=0x7fffeccf8ba0 <__pctx.7470>) at lib/assert.c:52
#4  0x00007fffec861fd4 in be_queue_op_get (successful=0x9218e8, data=0x7fffea0b1c40, op=0x9219e0, bq=0x8b8e60) at be/queue.c:299
#5  m0_be_queue_get (bq=bq@entry=0x8b8e60, op=op@entry=0x9219e0, data=data@entry=0x7fffea0b1c40, successful=successful@entry=0x9218e8) at be/queue.c:397
#6  0x00007fffec866d1b in be_tx_bulk_queue_get_cb (grp=<optimized out>, ast=<optimized out>) at be/tx_bulk.c:291
#7  0x00007fffec98e485 in m0_sm_asts_run (grp=grp@entry=0x74c4f0) at sm/sm.c:171
#8  0x00007fffec8d4677 in loc_handler_thread (th=0x74b0c0) at fop/fom.c:919
#9  0x00007fffec90333e in m0_thread_trampoline (arg=arg@entry=0x74b0c8) at lib/thread.c:117
#10 0x00007fffec90cabd in uthread_trampoline (arg=0x74b0c8) at lib/user_space/uthread.c:98
#11 0x00007fffec0c1e65 in start_thread () from /lib64/libpthread.so.0
#12 0x00007fffeb1b288d in clone () from /lib64/libc.so.6
```

Reply:
Right, because the # of consumers is the number of workers. The UT doesn't fail with the following patch:

```
> git diff
diff --git a/be/ut/tx_bulk.c b/be/ut/tx_bulk.c
index b69ce0e..f4a9ab0 100644
--- a/be/ut/tx_bulk.c
+++ b/be/ut/tx_bulk.c
@@ -204,9 +204,9 @@ void m0_be_ut_tx_bulk_usecase(void)
                .tbc_q_cfg                 = {
                        .bqc_q_size_max       = BE_UT_TX_BULK_Q_SIZE_MAX,
                        .bqc_producers_nr_max = 1,
-                       .bqc_consumers_nr_max = 0x100,  /* XXX */
+                       .bqc_consumers_nr_max = 1,
                },
-               .tbc_workers_nr            = 0x40, /* XXX */
+               .tbc_workers_nr            = 1, /* XXX */
                .tbc_partitions_nr         = 1,    /* XXX */
                .tbc_work_items_per_tx_max = 3,
                .tbc_do                    = &be_ut_tx_bulk_usecase_do,
```

Reply:
Made the # of workers configurable through a constant.